### PR TITLE
Reformat bot & Added confirmation/close to the help buttons.

### PR DESCRIPTION
**Pull request by Soheab**
This PR reformats some files because they sucked. Also adds confirmation to the help buttons when clicked the user will be asked if they really need help and a Close button in the thread for easy closing.

Please comment if anything is wrong or needs changing.

### CODE REVIEW

File: cogs/help.py [hunk 1]
```diff
@@ -1,63 +1,185 @@
...
+        self._help_type = help_type
+
+    async def create_help_thread(self, interaction: Interaction) -> None:
+        name = "Discord.py libraries" if self._help_type == "D.py libs" else self._help_type
+
+        thread = await interaction.channel.create_thread(
+            name = f"{name} help ({interaction.user})",
+            type = ChannelType.public_thread
+        )
+        # don't need this since the ping already adds them.
```
Comment (TAG-Epic) [inline cogs/help.py:23]: Dead code should be removed

File: cogs/help.py [hunk 2]
```diff
@@ -1,63 +1,185 @@
...
+
+        name_to_colour: Dict[str, Colour] = {
+            "Discord.py libraries": Colour.light_gray,
+            "Nextcord": Colour.red(),
+            "Python": Colour.green()
+        }
+        em = Embed(
+            title = f"{name} Help needed!",
+            description = f"Alright now that we are all here to help, "
+                          f"what do you need help with {interaction.user.mention}?",
```
Comment (TAG-Epic) [inline cogs/help.py:39]: Personally don't see a reason for the ping here.

File: cogs/help.py [hunk 3]
```diff
@@ -1,63 +1,185 @@
...
+            colour = name_to_colour[name]
+        )
+        em.set_footer(text = "Helpers or You can click the Close button to close this thread.")
```
Comment (TAG-Epic) [inline cogs/help.py:42]: I don't really like how this is worded. Close should probably be lowercase too?

File: cogs/help.py [hunk 4]
```diff
@@ -1,63 +1,185 @@
...
+    async def callback(self, interaction: Interaction):
+        self.view.value = True if self.custom_id == "confirm_button" else False
+        self.view.stop()
+
+
+class ConfirmView(ui.View):
+    def __init__(self):
+        super().__init__(timeout = 10.0)
+        self.value = None
+        self.add_item(ConfirmButton("Yes", ButtonStyle.green, custom_id = "confirm_button"))
```
Comment (TAG-Epic) [inline cogs/help.py:100]: Buttons for help should probably start with `help:...`

File: cogs/help.py [hunk 5]
```diff
@@ -1,63 +1,185 @@
...
     @commands.Cog.listener()
     async def on_message(self, message):
-        if message.channel.id == self.help_channel and message.type == MessageType.thread_created:
-            await message.delete(delay=5)
+        if message.channel.id == HELP_CHANNEL_ID and message.type is MessageType.thread_created:
+            await message.delete(delay = 5)
+        if isinstance(message.channel, Thread) and \
+                message.channel.parent_id == HELP_CHANNEL_ID and \
+                message.type is MessageType.pins_add:
+            await message.delete(delay = 10)
```
Comment (TAG-Epic) [inline cogs/help.py:162]: I don't see why we should delete pin messages here

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: cogs/help.py (hunk 5 above)
Comment (Soheab) [reply cogs/help.py:162]: The chat looks cleaner without imo. It's after 10 seconds to the user knows that it got pinned.

File: cogs/help.py (hunk 4 above)
Comment (Soheab) [reply cogs/help.py:100]: I don't see the custom_id of any button being used where so I assumed it didn't matter, will add the prefix.

File: cogs/help.py (hunk 3 above)
Comment (Soheab) [reply cogs/help.py:42]: I was thinking of "You and the helpers can close this thread with the button" ?

File: cogs/help.py (hunk 2 above)
Comment (Soheab) [reply cogs/help.py:39]: Same, removing.

File: cogs/help.py (hunk 3 above)
Comment (TAG-Epic) [reply cogs/help.py:42]: Sounds good

File: cogs/help.py (hunk 5 above)
Comment (TAG-Epic) [reply cogs/help.py:162]: Ah fair

File: cogs/help.py (hunk 1 above)
Comment (Soheab) [reply cogs/help.py:23]: Left it for review purposes for if you wanted it to be added again idk, will remove it.

**Review by TAG-Epic — APPROVED** [review]

**Pull request merged**